### ISSUE TITLE
Switching form Postgres to Prometheus in panel edit mode causes panel to disappear

### ISSUE THREAD

**Issue by tomgs** [grafana/grafana#18600]
**What happened**: Switching from Postgres To Prometheus as the source when editing a panel causes the entire editing window to disappear.

**What you expected to happen**: That the panel **will not** disappear.

**How to reproduce it (as minimally and precisely as possible)**: Make sure you have Promethues & Postgres as data sources, as well as a dashboard that has a panel that pulls from postgres. Click that panel's dropdown link , choose edit, then try switching to Prometheus as the data source.

**Anything else we need to know?**: n/a

**Environment**:
- Grafana version: 6.3.2
- Data source type & version: Prometheus & Postgres
- OS Grafana is installed on: Inside a Docker container with the official Grafana image
- User OS & Browser: Mac OS Mojave 
- Grafana plugins: None
- Others: N/A

**Comment by hugohaggmark** [comment]
Thanks for reporting this @tomgs, we'll have a looksy.

**Comment by hugohaggmark** [comment]
Hi again @tomgs, if you can reproduce it can you paste any Browser console errors here and which visualization are you using?

**Comment by tomgs** [comment]
Hey @hugohaggmark - here are the console errors:
```
react-dom.production.min.js:198 TypeError: Cannot read property 'length' of undefined
    at query_hints.ts:117
    at Array.forEach (<anonymous>)
    at query_hints.ts:116
    at t.getQueryHints (datasource.ts:711)
    at t.a.refreshHint (PromQueryField.tsx:279)
    at t.componentDidMount (PromLink.tsx:26)
    at Ha (react-dom.production.min.js:216)
    at Wa (react-dom.production.min.js:220)
    at react-dom.production.min.js:250
    at Object.t.unstable_runWithPriority (scheduler.production.min.js:18)
fa @ react-dom.production.min.js:198
query_hints.ts:117 Uncaught (in promise) TypeError: Cannot read property 'length' of undefined
    at query_hints.ts:117
    at Array.forEach (<anonymous>)
    at query_hints.ts:116
    at t.getQueryHints (datasource.ts:711)
    at t.a.refreshHint (PromQueryField.tsx:279)
    at t.componentDidMount (PromLink.tsx:26)
    at Ha (react-dom.production.min.js:216)
    at Wa (react-dom.production.min.js:220)
    at react-dom.production.min.js:250
    at Object.t.unstable_runWithPriority (scheduler.production.min.js:18)
```

And I'm using a Graph.

This is how the screen looks:

![image](https://user-images.githubusercontent.com/8835035/63174141-0a093800-c04a-11e9-83c1-2dadbd793c5e.png)

One more note: It does not happen if I set the panel's data source to Prometheus to begin with, i.e. when creating the panel. Only happens when moving from Postgres to Prometheus.

**Comment by hugohaggmark** [comment]
I've managed to repro this once but the other way around, going from prometheus to postgres but I've tried 50 time so haven't found what could cause this yet.

**Comment by hugohaggmark** [comment]
I can repro this:
1. Create new Dashboard
2. Add query
3. Select Postgres
4. Use `Select 1` as Query and change to `Format as Table`
![image](https://user-images.githubusercontent.com/562238/63243261-49a66e80-c25a-11e9-87ee-eee6c3291fce.png)
5. Select `Prometheus` datasource the whole screen turns black and console reports same as @tomgs reports above
![image](https://user-images.githubusercontent.com/562238/63243312-69d62d80-c25a-11e9-80c7-ac106ba50233.png)

**Comment by hugohaggmark** [comment]
@tomgs we'll try to have this fix in next patch release `6.3.4`

**Comment by tomgs** [comment]
@hugohaggmark Awesome, thanks!
